test/merge-patient-lab-iips-mocha.js

```diff
@@ -0,0 +1,138 @@
+                    return component.rdfpf.createVni('').getInputState('patient', '', '');
+                }).then(_.property('data')).should.become({id: '001',  name: 'Alice', dob: '1979-01-23' });
+            });
+            it("should have labwork input state after input", function() {
+                return Promise.resolve(component).then(function(component){
+                    var labwork = noflo.internalSocket.createSocket();
+                    component.inPorts.labwork.attach(labwork);
+                    labwork.send({id: '001',  glucose: '75',  date: '2012-02-01'});
+                    labwork.disconnect();
+                    component.inPorts.patient.detach(labwork);
+                    return component.rdfpf.createVni('').getInputState('labwork', '', '');
+                }).then(_.keys).then(_.sortBy).should.become(_.sortBy(['data','lm', 'previousLms', 'selfPort']));
+            });
+            it("should have patient input state data after input", function() {
+                return Promise.resolve(component).then(function(component){
+                    var labwork = noflo.internalSocket.createSocket();
+                    component.inPorts.labwork.attach(labwork);
+                    labwork.send({id: '001',  glucose: '75',  date: '2012-02-01'});
+                    labwork.disconnect();
+                    component.inPorts.patient.detach(labwork);
+                    return component.rdfpf.createVni('').getInputState('labwork', '', '');
+                }).then(_.property('data')).should.become({id: '001',  glucose: '75',  date: '2012-02-01'});
+            });
+        });
+        describe('#getOutputState', function() {
+            it("should initially return nothing", function() {
+                return Promise.resolve(component).then(function(component){
+                    return component.rdfpf.createVni('').getOutputState();
+                }).should.become(undefined);
+            });
+            it("should have a output state after input", function() {
+                return Promise.resolve(component).then(function(component){
+                    var patient = noflo.internalSocket.createSocket();
+                    var labwork = noflo.internalSocket.createSocket();
+                    component.inPorts.patient.attach(patient);
+                    patient.send({id: '001',  name: 'Alice', dob: '1979-01-23' });
+                    patient.disconnect();
+                    component.inPorts.labwork.attach(labwork);
+                    labwork.send({id: '001',  glucose: '75',  date: '2012-02-01'});
+                    labwork.disconnect();
+                    component.inPorts.patient.detach(labwork);
+                    component.inPorts.patient.detach(patient);
+                    return component.rdfpf.createVni('').getOutputState();
+                }).then(_.keys).then(_.sortBy).should.become(_.sortBy(['data','lm', 'previousLms', 'selfPort']));
```

Comment:
Nope - updater may not have run.   I believe this will be an issue with this test.